cmake/UnitTests.cmake

```diff
@@ -2358,6 +2357,36 @@ list(APPEND CPU_POCL_FAILED_TESTS "Unit_deviceFunctions_CompileTest_y1f_float")
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_deviceFunctions_CompileTest_ynf_float") # Failed
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipClassKernel_Friend") # SEGFAULT
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipDeviceSynchronize_Functional") # Failed
+# broken tests, they all try to write outside allocated memory;
+# valgrind + pocl shows:
+#
+#==11492== Invalid write of size 1
+#==11492==    at 0x5605B83: pocl_fill_aligned_buf_with_pattern (pocl_util.c:2590)
+#==11492==    by 0x562B1F5: pocl_driver_svm_fill (common_driver.c:444)
+#==11492==    by 0x5626EBF: pocl_exec_command (common.c:693)
+#==11492==    by 0x562205F: pthread_scheduler_get_work (pthread_scheduler.c:529)
+#==11492==    by 0x56221B7: pocl_pthread_driver_thread (pthread_scheduler.c:588)
+#==11492==    by 0x5009B42: start_thread (pthread_create.c:442)
+#==11492==    by 0x509ABB3: clone (clone.S:100)
+#==11492==  Address 0x114baf00 is 0 bytes after a block of size 16,384 alloc'd
+#==11492==    by 0x561EE67: pocl_basic_svm_alloc (basic.c:841)
+#==11492==    by 0x561411A: POclSVMAlloc (clSVMAlloc.c:98)
+#==11492==    by 0x4966168: SVMemoryRegion::allocate(unsigned long) (source/chip-spv/src/backend>
+#
+# running with older PoCL: "double free or corruption"
+#
+# running with Intel CPU runtime or new PoCL: hipErrorRuntimeMemory (CL_INVALID_VALUE )
+# in CHIPBackendOpenCL.cc:1048:memFillAsyncImpl
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2D_BasicFunctional") # Timeout
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_BasicFunctional") # Timeout
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_WithKernel") # Timeout
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_MultiThread") # Timeout
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetFunctional_ZeroValue_2D") # Timeout
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetASyncMulti") # Failed
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - int8_t") # Failed
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - int16_t") # Failed
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - uint32_t") # Failed
```

Comment:
Do I need to update POCL on the CI machine to get these to fail?

Reply:
Not necessary. The PR with extra checks hasn't been merged to PoCL's master, so the first 5 tests should fail with current PoCL master branch. After it's merged, it'll be the last 4 tests that should fail. However, since any one of them can fail or pass, depending on where their out-of-bounds access falls, i'd like them all disabled for PoCL. Unreliable tests are a waste of time.

Reply:
> so the first 5 tests should fail with current PoCL master branch

ok, so my question remains: what do I need to do to get them to fail on my machine?

> Unreliable tests are a waste of time.

right, that's why I keep pushing on this point. From what I understand about POCL these tests should fail everywhere not just on your machine. We should investigate why these fail for you and not on the CI machine and then make the necessary changes to the CI machine so that these fail universally. 

I want the CI machine to fail at least as much as any other machine the end user might have.

Reply:
> what do I need to do to get them to fail on my machine?
> ...
> We should investigate why these fail for you and not on the CI machine...

Out-of-bounds accesses are undefined behavior in C/C++. Anything can happen and the outcome may change just every time the program is launched.

You could try to catch OoB accesses by compiling and running the CI with [AddressSanitizer](https://clang.llvm.org/docs/AddressSanitizer.html).

Reply:
> Anything can happen and the outcome may change just every time the program is launched.

In this case I should be able to run it like 1000 times and get at least one of these tests to fail? 

I'm ok with excluding these tests - just trying to understand why these failures haven't popped up on the CI machine and if I need to do something to it to bring it more in-line with what end users might experience.

Reply:
> In this case I should be able to run it like 1000 times and get at least one of these tests to fail?

There is no guarantee a failure would occur. Anything can happen - including a hypothetical case where the test failure is impossible in some builds/systems but fails sometimes/always in other builds/systems. If an undefined behavior is triggered in C/C++ application, all bets are off.

Reply:
FWIW, i have tried compiling PoCL with different Debug/Release configurations, and different -Ox options. Some failed with a timeout, and some failed with a segfault. The most reliable way to see the bug is using valgrind, however sanitizers Henry mentioned should work too.